src/ObjectUtils/include/ObjectUtils/SymbolsFile.h

```diff
@@ -23,6 +23,25 @@ struct ObjectFileInfo {
   uint64_t executable_segment_offset = 0;
 };
 
+// Raw structure representing a function symbol.
+struct FunctionSymbol {
+  // Mangled symbol name. Can be empty if demangled name is available directly from the parser.
+  std::string mangled_name;
+  // Demangled symbol name.
+  std::string demangled_name;
+  // Relative virtual address, containing load bias and executable segment offset.
+  uint64_t relative_virtual_address = 0;
+  // Function code size in bytes.
+  uint32_t size = 0;
+};
```

Comment:
I think this should be a `class` with constructors. The style guide describes the difference between struct and classes via invariants:
> structs should be used for passive objects that carry data. [...] The struct must not have invariants that imply relationships between different fields.

I think there are invariants on this. Maybe "a valid SymbolInfo has at least a mangled_name or a demangled_name". Or: "A SymbolInfo with a demangled_name has always a mangled_name". Or "A valid SymbolInfo has always an address".

Right now its really hard to use FunctionSymbol, because wherever its used, (in theory) the user needs to check whether those fields are empty.